sdk/utils/utils.go

```diff
@@ -48,5 +50,6 @@ func GetAPILargeContext() (context.Context, context.CancelFunc) {
 
 // Timed Context
 func GetTimedContext(timeout time.Duration) (context.Context, context.CancelFunc) {
-	return context.WithTimeout(context.Background(), timeout)
+	parent, _ := signal.NotifyContext(context.Background(), os.Interrupt)
+	return context.WithTimeout(parent, timeout)
```

Comment:
You can avoid dropping the cancellation by wrapping the two together:

```suggestion
	parent, sigCancel := signal.NotifyContext(context.Background(), os.Interrupt)
	ctx, timeCancel := context.WithTimeout(parent, timeout)
	return ctx, func() {
		sigCancel()
		timeCancel()
	}()
```

Separately, but beyond the scope of this PR, typically `context.Background()` is reserved for `main()` and `init()` while all other functions should receive it. [More info](https://google.github.io/styleguide/go/decisions.html#:~:text=It%20is%20very%20rare%20for%20code%20in%20the%20middle%20of%20a%20callchain%20to%20require%20creating%20a%20base%20context%20of%20its%20own%20using%20context.Background().%20Always%20prefer%20taking%20a%20context%20from%20your%20caller%2C%20unless%20it%E2%80%99s%20the%20wrong%20context.).